@rebert 
it is recommended by [shellcheck](https://github.com/koalaman/shellcheck/wiki/SC2143) to replace:
``[[ -z $(psql -lqt -h localhost | egrep "\b${source_db}\b" 2> /dev/null) ]]``
with
``! psql -lqt -h localhost | egrep -q "\b${source_db}\b"``

In some cases there can be issues when using the pipefail bash option, see commit: https://github.com/geoadmin/deploy/pull/39/commits/1280e61d1b6aadcb1d1e20b1ea42d642583f2686

--- a/db_master_deploy/deploy.sh
+++ b/db_master_deploy/deploy.sh
@@ -366,7 +366,7 @@ copy_table() {
         offset=$(echo "((${i}-1)*${increment})" | bc)
         if [ $((offset+increment)) -gt "${rows}" ]; then counter=${rows}; else counter=$((offset+increment));fi
         echo "dumping ${offset}..${counter}"
-        ( psql -h localhost -qAt -d "${source_db}" -c "COPY ( SELECT ${columns} FROM ${source_schema}.${source_table} order by ${primary_keys:=1} asc offset ${offset} limit ${increment} ) TO STDOUT with csv" | psql -h localhost -qAt -d "${target_db}" -c "SET session_replication_role = replica; COPY ${target_schema}.${target_table} (${columns}) from stdin with csv; SET session_replication_role = DEFAULT;" )& pids+="$!"
+        ( psql -h localhost -qAt -d "${source_db}" -c "COPY ( SELECT ${columns} FROM ${source_schema}.${source_table} order by ${primary_keys:=1} asc offset ${offset} limit ${increment} ) TO STDOUT with csv" | psql -h localhost -qAt -d "${target_db}" -c "SET session_replication_role = replica; COPY ${target_schema}.${target_table} (${columns}) from stdin with csv; SET session_replication_role = DEFAULT;" )& pids+=("$!")
     done;
     wait "${pids[@]}" 2> /dev/null
     )

--- a/db_master_deploy/deploy.sh
+++ b/db_master_deploy/deploy.sh
@@ -69,6 +69,7 @@ source "${MY_DIR}/includes.sh"
 #######################################
 check_table() {
     # check if source table exists, views cannot be deployed
+    set +o pipefail
     if ! psql -lqt -h localhost -c "SELECT table_catalog||'.'||table_schema||'.'||table_name FROM information_schema.tables where lower(table_type) not like 'view'" -d "${source_db}" 2> /dev/null | egrep -q "\b${source_id}\b"; then
         echo "source table does not exist ${source_id} " >&2
         exit 1
@@ -79,7 +80,7 @@ check_table() {
         "target table does not exist ${target_id}." >&2
         exit 1
     fi
-
+    set -o pipefail
     # check if source and target table have the same structure (column name and data type)
     source_columns=$(psql -h localhost -d "${source_db}" -Atc "select column_name,data_type FROM information_schema.columns WHERE table_schema = '${source_schema}' AND table_name = '${source_table}' order by 1;")
     columns=$(psql -h localhost -d "${source_db}" -Atc "select column_name FROM information_schema.columns WHERE table_schema = '${source_schema}' AND table_name = '${source_table}';" | xargs | sed -e 's/ /,/g') # comma separted list of all attributes for order independent copy command

